pkg/e2e/verify/user_webhook.go

```diff
@@ -27,15 +27,390 @@ var _ = ginkgo.Describe(userWebhookTestName, func() {
+		// the user is using the SRE IdP but is not in one of the protected
+		// groups
+		ginkgo.It("osd-sre-cluster-admins cannot create/update protected redhat users with SRE IDP and no group", func() {
+			userName := util.RandomStr(5) + "@redhat.com"
+			identities := []string{"OpenShift_SRE:" + util.RandomStr(5)}
+			h.Impersonate(rest.ImpersonationConfig{
+				UserName: "test-user@redhat.com",
+				Groups: []string{
+					"osd-sre-cluster-admins",
+					"system:authenticated",
+					"system:authenticated:oauth",
+				},
+			})
+			user, err := createUser(userName, identities, []string{}, h)
+			defer func() {
+				h.Impersonate(rest.ImpersonationConfig{})
+				deleteUser(user.Name, h)
+			}()
+			Expect(err).To(HaveOccurred())
+		}, float64(viper.GetFloat64(config.Tests.PollingTimeout)))
+
+		// osd-sre-cluster-admins cannot create/update protected RH users if
+		// the user is in one of the protected groups but is not using the SRE
+		// IdP
+		ginkgo.It("osd-sre-cluster-admins cannot create/update protected redhat users with other IDP and RH group", func() {
+			userName := util.RandomStr(5) + "@redhat.com"
+			identities := []string{"otherIDP:testing_string"}
+			groups := []string{"osd-devaccess"}
+			// we need to add the username to the group before we create the user,
+			// because the user object cannot be created until the username is in
+			// the group
+			addUserToGroup(userName, groups[0], h)
+			h.Impersonate(rest.ImpersonationConfig{
+				UserName: "test-user@redhat.com",
+				Groups: []string{
+					"osd-sre-cluster-admins",
+					"system:authenticated",
+					"system:authenticated:oauth",
+				},
+			})
+			user, err := createUser(userName, identities, groups, h)
+			defer func() {
+				h.Impersonate(rest.ImpersonationConfig{})
+				deleteUser(user.Name, h)
+			}()
+			Expect(err).To(HaveOccurred())
+		}, float64(viper.GetFloat64(config.Tests.PollingTimeout)))
+
+		// osd-sre-cluster-admins can manage RH users if the user is both not
+		// using the SRE IdP and is not in one of the protected groups
+		ginkgo.It("osd-sre-cluster-admins can manage non-protected redhat users with other IDP and no group", func() {
+			userName := util.RandomStr(5) + "@redhat.com"
+			identities := []string{"otherIDP:testing_string"}
+			h.Impersonate(rest.ImpersonationConfig{
+				UserName: "test-user@redhat.com",
+				Groups: []string{
+					"osd-sre-cluster-admins",
+					"system:authenticated",
+					"system:authenticated:oauth",
+				},
+			})
+			user, err := createUser(userName, identities, []string{}, h)
+			defer func() {
+				h.Impersonate(rest.ImpersonationConfig{})
+				deleteUser(user.Name, h)
+			}()
+			Expect(err).NotTo(HaveOccurred())
+			err = deleteUser(user.Name, h)
+			Expect(err).NotTo(HaveOccurred())
+		}, float64(viper.GetFloat64(config.Tests.PollingTimeout)))
+
+		// osd-sre-cluster-admins can manage customer users
+		ginkgo.It("osd-sre-cluster-admins can manage customer users with other IDP and no group", func() {
```

Comment:
Can they also manage customer users with random groups? Just thinking of another case, not sure how important it is to add

Reply:
yes, absolutely. the important bits are whether the user is in the SRE groups or not. I can add a test case to prove this

Reply:
I'm meh on it. Won't block merging and the card even says this doesn't need to cover every case. It was just the first thing that came to mind when reading this message. Up to you if you want to add it